src/openrct2/world/Map.cpp

```diff
@@ -1539,6 +1545,103 @@ void map_update_tiles()
     }
 }
 
+void mapScheduleCalcHighestTileHeight()
+{
+    gCalcHighestTileHeightScheduled = true;
+}
+
+void mapTryCalcHighestTileHeight(bool fullCheck)
+{
+    if (!fullCheck && !gCalcHighestTileHeightScheduled)
```

Comment:
You dont need check ```gCalcHighestTileHeightScheduled``` if you can check for ```gMapHighestTileHeightLoopPosition < maxTileIndex```, simply setting ```gMapHighestTileHeightLoopPosition``` to 0 will start re-start the loop.